wordpress-mobile/WordPress-iOS
# Fix upload completion handling on close-without-save

Fixes: https://github.com/wordpress-mobile/gutenberg-mobile/issues/2956

### Description

This is an attempt to fix an issue when closing the editor with no unsaved changes, while uploads are in progress. In that case, the upload completion processors are invoked upon upload completion, and the media references in the post are never updated.

<s>This solution has side effects, and needs more work to address that before proceeding. With these changes, the upload completion issue is resolved, but a new issue exists where re-opening the saved post, and closing it without making any changes still prompts the user to save unsaved changes.</s>

It turns out that the issue described above is already present, and not introduced by this PR.

To test:

* Add a media-containing block to a post (Image block is sufficient)
* Add media from device (a single image is enough)
* While media is uploading, exit the editor (saving / updating post when prompted)
* Open the post in the editor again (it is _not necessary_ to wait until 100% upload progress as described previously)
* Close the editor
* Allow the media to finish uploading
  * Observe the upload progress on the post card in the post list

The media should upload successfully. After re-opening the post (once media is finished uploading), the `src` references in the images should be the remote URL, not the local ones (you can verify via HTML mode).

PR submission checklist:
